.github/workflows/symbols-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Symbols Check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number && format('pr{0}', github.event.number) || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  Run:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 10
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          # So we can determine the merge base
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get build-dep ./
+
+      - name: Configure
+        run: >
+          cmake -B build ${{ github.workspace }}
+
+      - name: Install clang
+        run: |
+          wget https://apt.llvm.org/llvm.sh
```

Comment:
Hm. What are the security concerns with this `curl | sudo sh -`?

It's in the GitHub runners infrastructure, so infrastructure escape shouldn't be a problem, but what does it have access to *here*?

I think the answer is that it's got access to the GitHub secrets? Which means it has access to the release signing key, and the signing key to upload to the PPA?

I don't think this should be a blocker here, but maybe we should record this somewhere?